File: web/src/main/java/top/emilejones/hhu/web/service/strategy/RecallStrategy.java
package top.emilejones.hhu.web.service.strategy;

import kotlin.Pair;
import top.emilejones.hhu.web.entity.FileNode;
import top.emilejones.hhu.web.entity.TextNode;

import java.util.List;

public interface RecallStrategy {
    List<Pair<FileNode, TextNode>> exec(List<Pair<FileNode, TextNode>> rawData);
}

File: infra-textsplitter/src/main/kotlin/top/emilejones/hhu/service/strategy/RecallStrategy.kt
package top.emilejones.hhu.service.strategy

import top.emilejones.hhu.domain.po.Neo4jFileNode
import top.emilejones.hhu.domain.po.Neo4jTextNode

interface RecallStrategy {
    fun exec(rawData: List<Pair<Neo4jFileNode, Neo4jTextNode>>): List<Pair<Neo4jFileNode, Neo4jTextNode>>
}
